src/htmx.js

```diff
@@ -1923,10 +1919,13 @@ return (function () {
 
         function findElementsToProcess(elt) {
             if (elt.querySelectorAll) {
-                var boostedElts = hasChanceOfBeingBoosted() ? ", a" : "";
-                var results = elt.querySelectorAll(VERB_SELECTOR + boostedElts + ", form, [type='submit'], [hx-sse], [data-hx-sse], [hx-ws]," +
+                var boostedElts = document.querySelectorAll("[hx-boost] a, [data-hx-boost] a");
+                var results = elt.querySelectorAll(VERB_SELECTOR + ", form, [type='submit'], [hx-sse], [data-hx-sse], [hx-ws]," +
                     " [data-hx-ws], [hx-ext], [data-hx-ext], [hx-trigger], [data-hx-trigger], [hx-on], [data-hx-on]");
-                return results;
+                return [
+                    ...results,
+                    ...boostedElts,
```

Comment:
Note that the spread operator is [not IE11 compatible](https://caniuse.com/?search=spread%20operator), and htmx (1, IE11 support will be dropped for htmx 2) strives for IE11 compatibility